server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -410,16 +409,24 @@ static ClusterState closeRoutingTable(final ClusterState currentState,
                 }
 
                 logger.debug("closing index {} succeeded", index);
-                blocks.removeIndexBlockWithId(index.getName(), INDEX_CLOSED_BLOCK_ID).addIndexBlock(index.getName(), INDEX_CLOSED_BLOCK);
-                metadata.put(IndexMetaData.builder(indexMetaData).state(IndexMetaData.State.CLOSE));
-                routingTable.remove(index.getName());
-                closedIndices.add(index.getName());
+                final IndexMetaData updatedIndexMetaData = IndexMetaData.builder(indexMetaData).state(IndexMetaData.State.CLOSE).build();
+                metadata.put(updatedIndexMetaData, true);
+
+                blocks.removeIndexBlockWithId(index.getName(), INDEX_CLOSED_BLOCK_ID);
+                blocks.addIndexBlock(index.getName(), INDEX_CLOSED_BLOCK);
+
+                closedIndices.add(index);
             } catch (final IndexNotFoundException e) {
                 logger.debug("index {} has been deleted since it was blocked before closing, ignoring", index);
             }
         }
+
+        final ClusterState updatedState = ClusterState.builder(currentState).metaData(metadata).blocks(blocks).build();
+        final RoutingTable.Builder routingTable = RoutingTable.builder(updatedState.routingTable());
+        closedIndices.forEach(index -> routingTable.addAsFromOpenToClose(updatedState.metaData().getIndexSafe(index)));
```

Comment:
why update routing table here instead of above like it used to be? The previous way looked simpler I think

Reply:
I reverted this change (I found it easier to read in the other way)